### ISSUE TITLE
PyTorch conda linux binares should be build with CuSparseLT support

### ISSUE THREAD

**Issue by malfet** [pytorch/pytorch#115085]
### 🐛 Describe the bug

For the sake of uniformity with Linux CUDA wheels

### Versions

All

cc @ezyang @gchanan @zou3519 @kadeng @seemethere @osalpekar @atalman @alexsamardzic @nikitaved @pearu @cpuhrsch @amjames @bhosmer

**Comment by malfet** [comment]
Marked this one as hi-pri as sparse team expects this to be available for 2.2 , but at the same time we should have a better process for adding third-party dependencies to the project.

**Comment by malfet** [comment]
Tested that its available in latest nightly:
```
conda create -n py311 pytorch pytorch-cuda=12.1 python==3.11 -c nvidia -c pytorch-nightly
conda run -n py311 python -c "import torch;print(torch.__version__)"
2.3.0.dev20231220
$ conda run -n py311 python -c "import torch;print(open('/proc/self/maps').read())"|grep cusparseLt
7f5ce7398000-7f5ce90c2000 r-xp 00000000 103:08 16308572                  /home/nshulga/miniconda3/envs/py311/lib/python3.11/site-packages/torch/lib/libcusparseLt-f8b4a9fb.so.0
```

**Comment by huydhn** [comment]
This issue has been fixed in the upcoming 2.2.0 release.  I have tried both

```
python -c "import torch;print(open('/proc/self/maps').read())"|grep cusparseLt
7ff4ab600000-7ff4ad32a000 r-xp 00000000 00:26 371923983                  /home/huydo/py3.10/lib/python3.10/site-packages/torch/lib/libcusparseLt-f8b4a9fb.so.0
7ff4ad32a000-7ff4ad52a000 ---p 01d2a000 00:26 371923983                  /home/huydo/py3.10/lib/python3.10/site-packages/torch/lib/libcusparseLt-f8b4a9fb.so.0
7ff4ad52a000-7ff4ad72a000 r--p 01d2a000 00:26 371923983                  /home/huydo/py3.10/lib/python3.10/site-packages/torch/lib/libcusparseLt-f8b4a9fb.so.0
7ff4ad72a000-7ff4ad74a000 rw-p 01f2a000 00:26 371923983                  /home/huydo/py3.10/lib/python3.10/site-packages/torch/lib/libcusparseLt-f8b4a9fb.so.0
7ff4ad78e000-7ff4ad93c000 rw-p 01f4b000 00:26 371923983                  /home/huydo/py3.10/lib/python3.10/site-packages/torch/lib/libcusparseLt-f8b4a9fb.so.0
```

and 

```
import torch
from torch.sparse import to_sparse_semi_structured

A = torch.Tensor([0, 0, 1, 1]).tile((128, 32)).half().cuda()
A_sparse = to_sparse_semi_structured(A)
```

from https://github.com/pytorch/pytorch/issues/115077.  Both are working.
